traits/observers/_list_change_event.py

```diff
@@ -0,0 +1,72 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Event object for representing mutations to a list.
+"""
+
+# ListChangeEvent is exposed in the public API
+
+
+class ListChangeEvent:
+    """ Event object to represent mutations to a list.
+
+    Attributes
+    ----------
+    trait_list : list
+        The list being mutated.
+    index : int or slice
+        The index used for the mutation.
+    added : list
+        Values added to the list.
+    removed : list
+        Values removed from the list.
+    """
+
+    def __init__(self, *, trait_list, index, removed, added):
+        self.trait_list = trait_list
+        self.added = added
+        self.removed = removed
+        self.index = index
+
+    def __repr__(self):
+        return (
+            "<ListChangeEvent("
+            "trait_list={trait_list!r}, index={index!r}, "
+            "removed={removed!r}, added={added!r}"
+            ")>".format(
+                trait_list=self.trait_list,
+                index=self.index,
+                removed=self.removed,
+                added=self.added,
+            )
+        )
+
+
+def list_event_factory(trait_list, index, removed, added):
```

Comment:
I'd be tempted to make this a `classmethod` of `ListChangeEvent`

Reply:
Actually, on second thought, perhaps embrace the ordering and make `ListChangeEvent` a named tuple.

Reply:
The `__init__` of `ListChangeEvent` is currently the same as the call signature of `TraitList.notify` now, but they could evolve independently of each other.  `DictChangeEvent` in #1072 is actually an example where the `__init__` of `DictChangeEvent` is not the same as the call signature of `TraitDict.notify`.

The only reason why this is not a classmethod is to avoid exposing this to the user. `ListChangeEvent` is exposed in the public API via the public module `events` whereas the event factory isn't. An alternative organization would be to have all the public facing event object defined in a public module, and these event factories defined in another private modules.

Reply:
> The __init__ of ListChangeEvent is currently the same as the call signature of TraitList.notify

That's not true because of the keyword arguments...and of course that's why you asked in the next comment :sweat_smile: